sql/dd/impl/bootstrap/bootstrapper.cc

```diff
@@ -82,11 +82,30 @@ namespace {
 // Initialize recovery in the DDSE.
 bool DDSE_dict_recover(THD *thd, dict_recovery_mode_t dict_recovery_mode,
                        uint version) {
-  handlerton *ddse = get_dd_engine(thd);
-  if (ddse->dict_recover == nullptr) return true;
+  handlerton *ddse = nullptr;
 
+  if (opt_initialize) {
+    ddse = get_dd_engine(thd);
+  } else {
+    ddse = ha_resolve_by_legacy_type(
+        thd, bootstrap::DD_bootstrap_ctx::instance().get_actual_dd_engine());
+  }
+
+  if (ddse->dict_recover == nullptr) return true;
   bool error = ddse->dict_recover(dict_recovery_mode, version);
+  if (error) return error;
 
+  /*
+    Always call innodb handler API unless disabled
```

Comment:
the "unless disabled" part looks incorrect, until full conversion of system tables. InnoDB needs to be initialized always, because the system tables are still in it.
Same comment for `DDSE_dict_init` below

Reply:
Yes. We could only disable innodb after all tables has been move to rocksdb(including these system tables). that disabled check is helpful for future upgrade all tables to rocksdb development, such as easy to know which table still need innodb after we disable innodb..